### PR TITLE
Add Units to Cached Models

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This patch adds units to cached models, with deltas maintained by the model-cache worker.

## QA steps

New passing unit tests.

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Add Units to Model Cache](https://trello.com/c/49Xfx7PY/1064-add-units-to-model-cache)

File: core/cache/changes.go [hunk 1]
```diff
@@ -48,3 +48,30 @@ type RemoveApplication struct {
...
+type UnitChange struct {
+	ModelUUID      string
+	Name           string
+	Application    string
+	Series         string
+	CharmURL       string
+	PublicAddress  string
+	PrivateAddress string
+	MachineId      string
+	// TODO (manadart 2019-02-11): How should we handle these given that
```
Comment (manadart) [inline core/cache/changes.go]: @howbazaar What should we do here? 

I suppose there is a case for moving the network package under core insofar as it does not have other transient juju dependencies. Perhaps gradually?

File: core/cache/model.go [hunk 2]
```diff
@@ -62,7 +65,19 @@ func (m *Model) Application(appName string) (*Application, error) {
 		return nil, errors.NotFoundf("application %q", appName)
 	}
 	return app, nil
+}
 
+// Unit returns the unit for the input identifier.
+// If the unit is not found, a NotFoundError is returned.
+func (m *Model) Unit(unitId string) (*Unit, error) {
```
Comment (hmlanigan) [inline core/cache/model.go]: For consistency, isn't unitId really unitName?  If not, what's the difference between the identifier and a unit name?

Comment (manadart) [reply core/cache/model.go]: Fair point; I've renamed it.

**Comment by manadart** [comment]
$$merge$$

**Comment by howbazaar** [comment]
FAIL: package_test.go:57: ImportSuite.TestImports
```
package_test.go:61:
    // This package only brings in other core packages.
    c.Assert(found, jc.SameContents, []string{
        "core/constraints",
        "core/instance",
        "core/life",
        "core/status",
    })
... obtained []string = []string{"core/constraints", "core/instance", "core/life", "core/network", "core/status"}
... expected []string = []string{"core/constraints", "core/instance", "core/life", "core/status"}
```

Added `core/network`

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
Mongo start failure. $$merge$$

**Pull request merged**